src/lib/components/DeckGLMap/layers/piechart/pieChartLayer.ts

```diff
@@ -0,0 +1,130 @@
+import { SolidPolygonLayer, SolidPolygonLayerProps } from "@deck.gl/layers";
+import { COORDINATE_SYSTEM } from "@deck.gl/core";
+
+const defaultProps = {
+    pickable: true,
+    coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
+    getFillColor: (d) => d.properties.color,
+    getPolygon: (d) => d.polygon,
+};
+
+export type PieChartLayerProps<D> = SolidPolygonLayerProps<D>;
+export default class PieChartLayer extends SolidPolygonLayer<
+    unknown,
+    PieChartLayerProps<unknown>
+> {
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    shouldUpdateState({ changeFlags }) {
+        return changeFlags.viewportChanged;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    updateState({ props, oldProps, context, changeFlags }) {
+        super.updateState({ props, oldProps, context, changeFlags });
+
+        const zoom = context?.viewport?.zoom ?? 1;
+
+        if (props.data?.pies !== undefined) {
+            // Data is read and it is in proprietary format
+            // We need to store this data. Zoom is realtive to it.
+            // Store in new prop 'data_orig'
```

Comment:
This approach will not work anymore and it's not ideal to set new props from inside the prop anyway. Not sure yet what would be a good alternative.